src/bokeh/sampledata/__init__.py

```diff
@@ -50,22 +20,22 @@
 # Import
 #-----------------------------------------------------------------------------
 
-# isort: skip_file
+from typing import Any
+
+from packaging.version import Version
 
 #-----------------------------------------------------------------------------
 # Globals and constants
 #-----------------------------------------------------------------------------
 
-__all__ = (
-    'download',
-)
+__all__ = ()
+
+SAMPLEDATA_MIN_VERSION = "2024.2"
 
 #-----------------------------------------------------------------------------
 # General API
 #-----------------------------------------------------------------------------
 
-from ..util.sampledata import download
```

Comment:
I have used `download` in the Holoviz CI. So maybe have something like this (the language can change as needed):

``` python
def download():
    from warnings import warn
    warn("download is no longer needed, as all data is available in the `bokeh_sampledata` package", BokehUserWarning) 
```

This is mainly for people who have already installed the new package and don't understand why they can no longer download the data.

Reply:
https://github.com/bokeh/bokeh/pull/13874/commits/094987a8e11b31157f7e3ba6582599bb0b8550e0